### PR TITLE
DFT_embedding

### PR DESCRIPTION
**Pull request by sundaramvivek10**
Modified Hamiltonian for DFT embedding scheme

### CODE REVIEW

File: src/libxtp/pmdecomposition.cc [hunk 1]
```diff
@@ -0,0 +1,122 @@
+#include "votca/xtp/pmdecomposition.h"
```
Comment (JensWehner) [inline src/libxtp/pmdecomposition.cc:1]: Add license

File: share/xtp/xml/decomp.xml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0"?>
```
Comment (JensWehner) [inline share/xtp/xml/decomp.xml:1]: Please add proper explanation here.

File: src/libxtp/pmdecomposition.cc [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+//   return a;
+// }
+
+// Eigen::MatrixXd PMDecomposition::rowwise(const Eigen::MatrixXd &S, Eigen::VectorXd &v) {
+//   Eigen::MatrixXd a(S.rows(), S.cols());
+//   for (int p = 0; p < S.rows(); p++) {
+//     a.row(p) = v(p) * S.row(p);
+//   }
+//   return a;
+// }
```
Comment (JensWehner) [inline src/libxtp/pmdecomposition.cc:54]: remove commented out code pls

File: src/libxtp/pmdecomposition.cc [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+Eigen::MatrixXd PMDecomposition::rotatedorbitals(Eigen::MatrixXd &maxorbs, Index s, Index t) {
+  Eigen::VectorXd vec1, vec2, new_vec1, new_vec2;
+  double gam, sin_gamma, cos_gamma;
```
Comment (JensWehner) [inline src/libxtp/pmdecomposition.cc:58]: declare variables when they are used not before.

File: src/libxtp/tools/decomp.cc [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  orbitals.ReadFromCpt(_job_name + ".orb");
+}
+
+bool Decomp::Run() {
+  log.setReportLevel(Log::current_level);
+  log.setMultithreading(true);
+  log.setCommonPreface("\n... ...");
+  XTP_LOG(Log::error,log) << "Starting decomp tool" << std::endl;
+  PMDecomposition pmd(orbitals, log);
+  pmd.compute();
```
Comment (JensWehner) [inline src/libxtp/tools/decomp.cc:44]: I would put the orbitals arguement in the `compute()` method

**Review by JensWehner — CHANGES_REQUESTED** [review]
Nice start. I just skimmed over it. I think the most important things are a) documentation and b) tests that are lacking atm

File: src/libxtp/pmdecomposition.cc (hunk 1 above)
Comment (sundaramvivek10) [reply src/libxtp/pmdecomposition.cc:1]: Okay wait...I think I messed up a little with the branches. New user mistakes. This thing was supposed to be comitted into the decomptool branch. However I will incorporate these suggestions there

Comment (JensWehner) [reply src/libxtp/pmdecomposition.cc:1]: No problem, if you need help let me know.

**Comment by junghans** [comment]
Replaced by https://github.com/votca/votca/pull/846

**Pull request closed without merging**